normandy/classifier/models.py

```diff
@@ -2,22 +2,24 @@
 
 import uuid
 
-from django.utils.functional import cached_property
-
 from normandy.classifier.geolocation import get_country_code
 from normandy.recipes.models import Recipe
 
 
 class Client(object):
     """A client attempting to fetch a set of recipes."""
-    def __init__(self, request, locale=None):
+    def __init__(self, request, locale=None, release_channel=None, country=None):
         self.request = request
         self.locale = locale
+        self.release_channel = release_channel
+        self._country = country
```

Comment:
You can actually just set `self.country = country` and it will use that instead of the cached property. So setting it if country isn't `None` will let you retain the decorator while allowing it to be set manually.